src/lib/annotations/doc/DocDrawingThread.js

```diff
@@ -88,6 +98,132 @@ class DocDrawingThread extends DrawingThread {
             this.pendingPath = null;
         }
     }
+
+    /**
+     * Determine if the drawing in progress if a drawing goes to a different page
+     *
+     * @param {Object} location - The current event location information
+     * @return {boolean} Whether or not the thread page has changed
+     */
+    hasPageChanged(location) {
+        return this.location && this.location.page && this.location.page !== location.page;
+    }
+
+    /**
+     * Saves a drawing annotation to the drawing annotation layer canvas.
+     *
+     * @param {string} type - Type of annotation
+     * @param {string} text - Text of annotation to save
+     * @return {void}
+     */
+    saveAnnotation(type, text) {
+        super.saveAnnotation(type, text);
+        this.reset();
+
+        const drawingAnnotationLayerContext = docAnnotatorUtil.getContext(
+            this.pageEl,
+            CLASS_ANNOTATION_LAYER_DRAW,
+            PAGE_PADDING_TOP,
+            PAGE_PADDING_BOTTOM
+        );
+
+        if (drawingAnnotationLayerContext) {
+            const memoryCanvas = this.drawingContext.canvas;
+            drawingAnnotationLayerContext.drawImage(memoryCanvas, 0, 0);
+            this.drawingContext.clearRect(0, 0, memoryCanvas.width, memoryCanvas.height);
+        }
+    }
+
+    /**
+     * Display the document drawing thread. Will set the drawing context if the scale has changed since the last show.
+     *
+     * @return {void}
+     */
+    show() {
+        if (!this.annotatedElement || !this.location) {
+            return;
+        }
+
+        // Get DrawingPath objects to be shown
+        const drawings = this.getDrawings();
+        if (this.pendingPath && !this.pendingPath.isEmpty()) {
+            drawings.push(this.pendingPath);
+        }
+
+        this.checkAndHandleScaleUpdate();
+        // Get the annotation layer context to draw with
+        let context;
+        if (this.state === STATES.pending) {
+            context = this.drawingContext;
+        } else {
+            this.pageEl = docAnnotatorUtil.getPageEl(this.annotatedElement, this.location.page);
```

Comment:
show() is called when rescaling is done as well as when a drawing thread is initially created. This control block determines whether to draw the drawing to the in-progress context or to the commited drawing context.